flake.nix

```diff
@@ -0,0 +1,35 @@
+{
+  description = "A Nix-flake-based development environment for vscode-parse-tree";
+
+  inputs.nixpkgs.url = "github:NixOS/nixpkgs/nixpkgs-unstable";
+
+  outputs = { self, nixpkgs }:
+    let
+      overlays = [
+        (final: prev: rec {
+          nodejs = prev.nodejs-18_x;
+          yarn = (prev.yarn.override { inherit nodejs; });
+        })
+      ];
+      supportedSystems = [ "x86_64-linux" "aarch64-linux" "x86_64-darwin" "aarch64-darwin" ];
+      forEachSupportedSystem = f: nixpkgs.lib.genAttrs supportedSystems (system: f {
+        pkgs = import nixpkgs { inherit overlays system; config.allowUnfree = true; };
```

Comment:
Q: Do we need allowUnfree?

Reply:
I tried to load this up on my Mac but Nix is very unhappy with steam-run

<details>

```nix
> nix develop --show-trace
error:
       … while calling the 'derivationStrict' builtin

         at /builtin/derivation.nix:9:12: (source not available)

       … while evaluating derivation 'nix-shell'
         whose name attribute is located at /nix/store/rspism4p5xvwms84p5wfsnj93c87avmf-source/pkgs/stdenv/generic/make-derivation.nix:303:7

       … while evaluating attribute 'nativeBuildInputs' of derivation 'nix-shell'

         at /nix/store/rspism4p5xvwms84p5wfsnj93c87avmf-source/pkgs/stdenv/generic/make-derivation.nix:347:7:

          346|       depsBuildBuild              = lib.elemAt (lib.elemAt dependencies 0) 0;
          347|       nativeBuildInputs           = lib.elemAt (lib.elemAt dependencies 0) 1;
             |       ^
          348|       depsBuildTarget             = lib.elemAt (lib.elemAt dependencies 0) 2;

       … while evaluating derivation 'steam-run'
         whose name attribute is located at /nix/store/rspism4p5xvwms84p5wfsnj93c87avmf-source/pkgs/stdenv/generic/make-derivation.nix:303:7

       … while evaluating attribute 'buildCommand' of derivation 'steam-run'

         at /nix/store/rspism4p5xvwms84p5wfsnj93c87avmf-source/pkgs/build-support/trivial-builders/default.nix:87:14:

           86|       enableParallelBuilding = true;
           87|       inherit buildCommand name;
             |              ^
           88|       passAsFile = [ "buildCommand" ]

       … while evaluating derivation 'steam-run-bwrap'
         whose name attribute is located at /nix/store/rspism4p5xvwms84p5wfsnj93c87avmf-source/pkgs/stdenv/generic/make-derivation.nix:303:7

       … while evaluating attribute 'text' of derivation 'steam-run-bwrap'

         at /nix/store/rspism4p5xvwms84p5wfsnj93c87avmf-source/pkgs/build-support/trivial-builders/default.nix:146:16:

          145|     runCommand name
          146|       { inherit text executable checkPhase allowSubstitutes preferLocalBuild;
             |                ^
          147|         passAsFile = [ "text" ];

       … from call site

         at /nix/store/rspism4p5xvwms84p5wfsnj93c87avmf-source/pkgs/build-support/build-fhsenv-bubblewrap/default.nix:224:43:

          223|
          224|   bin = writeShellScript "${name}-bwrap" (bwrapCmd { initArgs = ''"$@"''; });
             |                                           ^
          225| in runCommandLocal name {

       … while calling 'bwrapCmd'

         at /nix/store/rspism4p5xvwms84p5wfsnj93c87avmf-source/pkgs/build-support/build-fhsenv-bubblewrap/default.nix:118:14:

          117|   indentLines = str: lib.concatLines (map (s: "  " + s) (filter (s: s != "") (lib.splitString "\n" str)));
          118|   bwrapCmd = { initArgs ? "" }: ''
             |              ^
          119|     ignored=(/nix /dev /proc /etc)

       … while evaluating derivation 'steam-run-fhs'
         whose name attribute is located at /nix/store/rspism4p5xvwms84p5wfsnj93c87avmf-source/pkgs/stdenv/generic/make-derivation.nix:303:7

       … while evaluating attribute 'buildCommand' of derivation 'steam-run-fhs'

         at /nix/store/rspism4p5xvwms84p5wfsnj93c87avmf-source/pkgs/build-support/trivial-builders/default.nix:87:14:

           86|       enableParallelBuilding = true;
           87|       inherit buildCommand name;
             |              ^
           88|       passAsFile = [ "buildCommand" ]

       … while evaluating derivation 'steam-run-usr-target'
         whose name attribute is located at /nix/store/rspism4p5xvwms84p5wfsnj93c87avmf-source/pkgs/stdenv/generic/make-derivation.nix:303:7

       … while evaluating attribute 'passAsFile' of derivation 'steam-run-usr-target'

         at /nix/store/rspism4p5xvwms84p5wfsnj93c87avmf-source/pkgs/build-support/trivial-builders/default.nix:88:7:

           87|       inherit buildCommand name;
           88|       passAsFile = [ "buildCommand" ]
             |       ^
           89|         ++ (derivationArgs.passAsFile or []);

       … while evaluating derivation 'ldconfig'
         whose name attribute is located at /nix/store/rspism4p5xvwms84p5wfsnj93c87avmf-source/pkgs/stdenv/generic/make-derivation.nix:303:7

       … while evaluating attribute 'text' of derivation 'ldconfig'

         at /nix/store/rspism4p5xvwms84p5wfsnj93c87avmf-source/pkgs/build-support/trivial-builders/default.nix:146:16:

          145|     runCommand name
          146|       { inherit text executable checkPhase allowSubstitutes preferLocalBuild;
             |                ^
          147|         passAsFile = [ "text" ];

       … from call site

         at /nix/store/rspism4p5xvwms84p5wfsnj93c87avmf-source/pkgs/stdenv/generic/check-meta.nix:449:16:

          448|         {
          449|           no = handleEvalIssue { inherit meta attrs; } { inherit (validity) reason errormsg; };
             |                ^
          450|           warn = handleEvalWarning { inherit meta attrs; } { inherit (validity) reason errormsg; };

       … while calling 'handleEvalIssue'

         at /nix/store/rspism4p5xvwms84p5wfsnj93c87avmf-source/pkgs/stdenv/generic/check-meta.nix:226:38:

          225|
          226|   handleEvalIssue = { meta, attrs }: { reason , errormsg ? "" }:
             |                                      ^
          227|     let

       error: Package ‘glibc-2.37-8’ in /nix/store/rspism4p5xvwms84p5wfsnj93c87avmf-source/pkgs/development/libraries/glibc/default.nix:173 is not available on the requested hostPlatform:
         hostPlatform.config = "aarch64-apple-darwin"
         package.meta.platforms = [
           "aarch64-linux"
           "armv5tel-linux"
           "armv6l-linux"
           "armv7a-linux"
           "armv7l-linux"
           "i686-linux"
           "loongarch64-linux"
           "m68k-linux"
           "microblaze-linux"
           "microblazeel-linux"
           "mips-linux"
           "mips64-linux"
           "mips64el-linux"
           "mipsel-linux"
           "powerpc64-linux"
           "powerpc64le-linux"
           "riscv32-linux"
           "riscv64-linux"
           "s390-linux"
           "s390x-linux"
           "x86_64-linux"
         ]
         package.meta.badPlatforms = [ ]
       , refusing to evaluate.

       a) To temporarily allow packages that are unsupported for this system, you can use an environment variable
          for a single invocation of the nix tools.

            $ export NIXPKGS_ALLOW_UNSUPPORTED_SYSTEM=1

        Note: For `nix shell`, `nix build`, `nix develop` or any other Nix 2.4+
        (Flake) command, `--impure` must be passed in order to read this
        environment variable.

       b) For `nixos-rebuild` you can set
         { nixpkgs.config.allowUnsupportedSystem = true; }
       in configuration.nix to override this.

       c) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
         { allowUnsupportedSystem = true; }
       to ~/.config/nixpkgs/config.nix.
```

</details>

Reply:
Ah I didn't realize steam-run was linux only, but makes sense I guess. I do have a nix-darwin system setup now so I'll be able to test stuff like that in the future. 

I also recently tested without using steam-run on nixos, when trying to upgrade tree-sitter-bash, and it seemed to work. I already dont recall why the old vendored web-tree-sitter seemed to need it, and the current vendored one doesnt. I vaguely recall it was something to do with building tree-sitter-haskell that was failing. Anyway, I'll remove steam-run entirely next time I poke at it. 

I started working on making it all buildable via the flake, using yarn2nix, but ran into issues with tree-sitter-hcl and didn't figure it out yet. Hopefully in the next week or two...

Reply:
Removed steam-run

Reply:
Do we still need allowUnfree then?

Reply:
Oops thanks. Removed